### PR TITLE
Created text to check user creation Post where user has Post.creator …

### PR DESCRIPTION
**Pull request by joelnv**
- Check user creation Post where user has Post.creator.
- Logged in user has /mine link  with posts.title  visible. whereas ,
- Anonymous user gets 404 page when he tries to visit  /mine link.

### CODE REVIEW

File: posts/tests.py [hunk 1]
```diff
@@ -155,3 +155,44 @@ def test_login_wrong_password(self):
...
+        self.signup1 = {'username': 'testuser2', 'password1': 'aaaa1234', 'password2': 'aaaa1234'}
+        self.login = {'username': 'testuser2', 'password': 'aaaa1234'}
+        User.objects.create_user(**self.login)
+        self.input = {'title':'What a super day', 'body':'this would have been a great day if...'}
+
+    def test_login_post_creation(self):
+        response = self.client.post(reverse('login'), self.login, follow=True)
+        self.assertTrue(response.context['user'].is_active)
+        response = self.client.post(reverse('posts:create'), self.input, follow=True)
+        instance = Post.objects.get()
```
Comment (sudheerkdty) [inline posts/tests.py]: Post.objects.get() will raise `MultipleObjectsReturned` error when there is more than a single Post.
Try to extract  id from the redirect url and use Post.object.get(id='')

Comment (sudheerkdty) [reply posts/tests.py]: And then validate fields of instance with input data

File: posts/tests.py [hunk 2]
```diff
@@ -155,3 +155,44 @@ def test_login_wrong_password(self):
...
+        self.assertRedirects(response, reverse('posts:post', kwargs={'id': instance.id}), status_code=302,
+                             target_status_code=200, fetch_redirect_response=True)
+        response = self.client.get(reverse('posts:myposts'), follow=True)
+        title = 'What a super day'
+        body = 'this would have been a great day if...'
+        self.assertContains(response, title, status_code=200)
+        self.assertNotContains(response, body, status_code=200)
```
Comment (sudheerkdty) [inline posts/tests.py]: You can test this within the previous method. `test_login_post_creation`

File: posts/tests.py [hunk 3]
```diff
@@ -155,3 +155,44 @@ def test_login_wrong_password(self):
...
+
+    def test_anonymous_user_has_no_mine_link(self):
+        response = self.client.post(reverse('login'), self.login, follow=True)
+        self.assertTrue(response.context['user'].is_active)
+        response = self.client.post(reverse('posts:create'), self.input, follow=True)
+        instance = Post.objects.get()
+        self.assertRedirects(response, reverse('posts:post', kwargs={'id': instance.id}), status_code=302,
+                             target_status_code=200, fetch_redirect_response=True)
```
Comment (sudheerkdty) [inline posts/tests.py]: what is the use of lines 189 - 194 in this function

Comment (joelnv) [reply posts/tests.py]: it was unnecessary , i thought of creating a post and then loggin oout and checking for '/mine' link
i have removed that in the coming pr

File: posts/tests.py [hunk 4]
```diff
@@ -155,3 +155,30 @@ def test_login_wrong_password(self):
...
+        response = self.client.post(reverse('login'), self.login, follow=True)
+        self.assertTrue(response.context['user'].is_active)
+        response = self.client.post(reverse('posts:create'), self.input, follow=True)
+        instance = Post.objects.get(id=1)
+        self.assertRedirects(response, reverse('posts:post', kwargs={'id': instance.id}), status_code=302,
+                             target_status_code=200, fetch_redirect_response=True)
+        response = self.client.get(reverse('posts:myposts'), follow=True)
+        title = 'What a super day'
+        body = 'this would have been a great day if...'
+        self.assertContains(response, title, status_code=200)
```
Comment (sudheerkdty) [inline posts/tests.py]: replace `title` with `self.input['title']` and `body` with `self.input['body']`

File: posts/tests.py [hunk 5]
```diff
@@ -155,3 +155,30 @@ def test_login_wrong_password(self):
...
+        self.signup1 = {'username': 'testuser2', 'password1': 'aaaa1234', 'password2': 'aaaa1234'}
+        self.login = {'username': 'testuser2', 'password': 'aaaa1234'}
+        User.objects.create_user(**self.login)
+        self.input = {'title':'What a super day', 'body':'this would have been a great day if...'}
+
+    def test_login_post_creation_and_checking_mine_link(self):
+        response = self.client.post(reverse('login'), self.login, follow=True)
+        self.assertTrue(response.context['user'].is_active)
+        response = self.client.post(reverse('posts:create'), self.input, follow=True)
+        instance = Post.objects.get(id=1)
```
Comment (sudheerkdty) [inline posts/tests.py]: extract `id` from `response.url`. hard coding wont work as expected always

**Pull request merged**